packages/sdks/snippets/gen1-nextjs/next.config.js

```diff
@@ -0,0 +1,22 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const path = require('path');
+
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  experimental: {
+    externalDir: true,
+  },
+  /**
+   * `useIsPreviewing` uses a different react version while linking so here are resolving it
+   * this is not required when using the SDK directly
+   */
+  webpack: (config) => {
+    config.resolve.alias['react'] = path.resolve(
+      __dirname,
+      './node_modules/react'
+    );
+    return config;
+  },
```

Comment:
had to add,
```
 "installConfig": {
    "hoistingLimits": "workspaces"
  },
```
and alias `react` package to the one present in the current `node_modules` as the `userIsPreviewing` hook was referencing a `useState` from a different version of React used by `@builder.io/react` causing issues